### PR TITLE
[FLINK-31525][tests] Stabilize JdbcExactlyOnceSinkE2eTest

### PR DESCRIPTION
**Pull request by eskabetxe**
## What is the purpose of the change

Stabilize JdbcExactlyOnceSinkE2eTest

## Brief change log

  - Remove job checkpoint configuration, default to cluster configuration
  - Change FailingMapper from random failure to expected failure, this allow to confirm the test behaviour

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-connectors/flink-connector-jdbc/src/test/java/org/apache/flink/connector/jdbc/xa/JdbcExactlyOnceSinkE2eTest.java [hunk 1]
```diff
@@ -195,22 +181,20 @@ public void testInsert() throws Exception {
...
-        int maxElementsPerFailure = numElementsPerCheckpoint * 3;
+        int numElementsPerCheckpoint = 10;
+        int expectedFailures = elementsPerSource / numElementsPerCheckpoint;
 
         StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
         env.setParallelism(dbEnv.getParallelism());
-        env.setRestartStrategy(fixedDelayRestart(Integer.MAX_VALUE, Time.milliseconds(100)));
-        env.setStreamTimeCharacteristic(TimeCharacteristic.ProcessingTime);
+        env.setRestartStrategy(fixedDelayRestart(expectedFailures * 2, Time.milliseconds(100)));
+        env.getConfig().setAutoWatermarkInterval(0L);
```
[the remark this answers is not in the document]
Comment (snuyanzin) [reply flink-connectors/flink-connector-jdbc/src/test/java/org/apache/flink/connector/jdbc/xa/JdbcExactlyOnceSinkE2eTest.java:190]: sounds reasonable, thanks for explanation

File: flink-connectors/flink-connector-jdbc/src/test/java/org/apache/flink/connector/jdbc/xa/JdbcExactlyOnceSinkE2eTest.java [hunk 2]
```diff
@@ -92,25 +89,14 @@
...
 @RunWith(Parameterized.class)
 public class JdbcExactlyOnceSinkE2eTest extends JdbcTestBase {
-    private static final Random RANDOM = new Random(System.currentTimeMillis());
-
     private static final Logger LOG = LoggerFactory.getLogger(JdbcExactlyOnceSinkE2eTest.class);
 
-    private static final long CHECKPOINT_TIMEOUT_MS = 20_000L;
-    private static final long TASK_CANCELLATION_TIMEOUT_MS = 20_000L;
-
-    // todo: remove after fixing FLINK-22889
```
Comment (snuyanzin) [inline flink-connectors/flink-connector-jdbc/src/test/java/org/apache/flink/connector/jdbc/xa/JdbcExactlyOnceSinkE2eTest.java:107]: I notice 2 more lines like about FLINK-22889 with todo remove after fixing.
Should they also be cleaned?

Comment (eskabetxe) [reply flink-connectors/flink-connector-jdbc/src/test/java/org/apache/flink/connector/jdbc/xa/JdbcExactlyOnceSinkE2eTest.java:107]: yes removed

**Comment by snuyanzin** [comment]
lgtm, once ci is passed

**Comment by eskabetxe** [comment]
@flinkbot run azure

**Comment by snuyanzin** [comment]
based on exception this should help https://github.com/apache/flink/commit/36f39712d289b6b7b0b957a33bd1891266e3f992

probably need to rebase

**Comment by eskabetxe** [comment]
@flinkbot run azure

**Comment by eskabetxe** [comment]
@flinkbot run azure

**Review by snuyanzin — APPROVED** [review]

**Pull request merged**